hack/tools/tilt-prepare/main.go

```diff
@@ -0,0 +1,486 @@
+		var stdout2, stderr2 bytes.Buffer
+		cmd2.Dir = rootPath
+		cmd2.Stdin = bytes.NewReader(stdout1.Bytes())
+		cmd2.Stdout = &stdout2
+		cmd2.Stderr = &stderr2
+		if err := cmd2.Run(); err != nil {
+			errCh <- fmt.Errorf("[%s] failed to run %s: %v, %s", prefix, cmd2.Args, err, stderr2.Bytes())
+			return
+		}
+
+		objs, err := utilyaml.ToUnstructured(stdout2.Bytes())
+		if err != nil {
+			errCh <- fmt.Errorf("[%s] failed parse components yaml: %v", prefix, err)
+			return
+		}
+
+		if debug {
+			if err := prepareDeploymentForDebug(prefix, objs); err != nil {
+				errCh <- err
+				return
+			}
+		}
+
+		providerObj, err := getProviderObj(prefix, objs)
+		if err != nil {
+			errCh <- err
+			return
+		}
+		objs = append(objs, *providerObj)
+
+		yaml, err := utilyaml.FromUnstructured(objs)
+		if err != nil {
+			errCh <- fmt.Errorf("[%s] failed convert untructure objects to yaml: %v", prefix, err)
+			return
+		}
+
+		if err := writeIfChanged(prefix, filepath.Join(tiltBuildPath, "yaml", out), yaml); err != nil {
+			errCh <- err
+		}
+	}
+}
+
+// writeIfChanged writes yaml to a file if the filed does not exist or if the content is changed.
+// NOTE: Skipping write in case the content is not changed avoids unnecessary Tiltfile reload.
+func writeIfChanged(prefix string, path string, yaml []byte) error {
+	_, err := os.Stat(path)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("[%s] failed to check if %s exists: %v", prefix, path, err)
+	}
+
+	if err == nil {
+		previousYaml, err := os.ReadFile(path) //nolint:gosec
+		if err != nil {
+			return fmt.Errorf("[%s] failed to read existing %s: %v", prefix, path, err)
+		}
+		if bytes.Equal(previousYaml, yaml) {
+			fmt.Printf("[%s] no changes in the generated yaml\n", prefix)
+			return nil
+		}
+	}
+
+	err = os.MkdirAll(filepath.Dir(path), 0750)
+	if err != nil {
+		return fmt.Errorf("[%s] failed to create dir %s: %v", prefix, filepath.Dir(path), err)
+	}
+
+	if err := os.WriteFile(path, yaml, 0600); err != nil {
+		return fmt.Errorf("[%s] failed to write %s: %v", prefix, path, err)
+	}
+	return nil
+}
+
+func prepareDeploymentForDebug(prefix string, objs []unstructured.Unstructured) error {
+	for i := range objs {
+		obj := objs[i]
+		if obj.GetKind() != "Deployment" {
+			continue
+		}
+
+		// Convert Unstructured into a typed object
+		d := &appsv1.Deployment{}
+		if err := scheme.Scheme.Convert(&obj, d, nil); err != nil {
+			return fmt.Errorf("[%s] failed to convert Deployment to typed object: %v", prefix, err)
+		}
+
+		for j := range d.Spec.Template.Spec.Containers {
+			container := d.Spec.Template.Spec.Containers[j]
+
+			// Drop liveness and readiness probes.
+			container.LivenessProbe = nil
+			container.ReadinessProbe = nil
+
+			// Drop leader election.
+			debugArgs := make([]string, 0, len(container.Args))
+			for _, a := range container.Args {
+				if a == "--leader-elect" { // make this more robust (e.g. detect --leader-elect=true)
```

Comment:
Should we add an `||` to check both? Or check prefix instead?

Reply:
Is it also possible to specify it via two separate array elements?
```yaml
    - --leader-elect
    - true
```

Reply:
I have managed the two most common use cases, then we can eventually iterate